### PR TITLE
[backport] Test sparse with many types of sparse matrixes

### PR DESCRIPTION
**Pull request by niboshi**
Backport of #833

### CODE REVIEW

File: tests/cupy_tests/sparse_tests/test_csc.py [hunk 1]
```diff
@@ -426,305 +443,354 @@ def test_tocsr_copy(self, xp, sp):
...
         return m.transpose().toarray()
 
     @testing.numpy_cupy_raises(sp_name='sp', accept_error=ValueError)
     def test_transpose_axes_int(self, xp, sp):
-        m = _make(xp, sp, self.dtype)
+        m = self.make(xp, sp, self.dtype)
         m.transpose(axes=0)
 
+    @testing.numpy_cupy_equal(sp_name='sp')
+    def test_eliminate_zeros(self, xp, sp):
```
Comment (unnonouno) [inline tests/cupy_tests/sparse_tests/test_csc.py]: `eliminate_zeros` is not implemented in v2.

File: tests/cupy_tests/sparse_tests/test_csr.py [hunk 2]
```diff
@@ -449,300 +467,354 @@ def test_tocsr_copy(self, xp, sp):
...
+        m = self.make(xp, sp, self.dtype)
+        return m.transpose().toarray()
+
+    @testing.numpy_cupy_raises(sp_name='sp', accept_error=ValueError)
+    def test_transpose_axes_int(self, xp, sp):
+        m = self.make(xp, sp, self.dtype)
+        m.transpose(axes=0)
+
+    @testing.numpy_cupy_equal(sp_name='sp')
+    def test_eliminate_zeros(self, xp, sp):
```
Comment (unnonouno) [inline tests/cupy_tests/sparse_tests/test_csr.py]: ditto

**Comment by niboshi** [comment]
Remove tests of eliminate_zeros.

**Comment by unnonouno** [comment]
The test fails. Do you know the reason?

**Comment by niboshi** [comment]
Ah, is it because `sum` is not backported? #712

**Comment by kmaehashi** [comment]
This PR has been closed as `v2` branch is frozen.

**Pull request closed without merging**